README.md

```diff
@@ -1,69 +1,54 @@
-
-For more control and possible reuse of engine instances for multiple tests the module testing environment can be used explicitly. 
-This requires manual setup and tear down, but allows to run multiple tests against the same engine instance.
-
 For complete docs please see the
-[documentation on Github Pages](https://terasology.github.io/ModuleTestingEnvironment/org/terasology/moduletestingenvironment/ModuleTestingEnvironment.html)
+[documentation on Github Pages](https://terasology.github.io/ModuleTestingEnvironment/org/terasology/moduletestingenvironment/MTEExtension.html)
 
 For more examples see
 [the test suite](https://github.com/terasology/ModuleTestingEnvironment/tree/master/src/test/java/org/terasology/moduletestingenvironment)
 
 Here's an example taken from the test suite:
 
 ```java
-public class MyModuleEngineTest extends ModuleTestingEnvironment {
-    protected Logger logger = Logger.getLogger(MyModuleEngineTest.class.getName());
-    protected EntityManager entityManager;
-
-    @Override
-    public Set<String> getDependencies() {
-        return Sets.newHashSet("MyModule");
-    }
-
-    @Override
-    public String getWorldGeneratorUri() {
-        return "mymodule:mymodulesworldgenerator";
-    }
+@ExtendWith(MTEExtension.class)
+@Dependencies({"engine", "ModuleTestingEnvironment"})
+public class ExampleTest {
+
+    @In
+    private WorldProvider worldProvider;
+    @In
+    private BlockManager blockManager;
+    @In
+    private EntityManager entityManager;
+    @In
+    private ModuleTestingHelper helper;
 
-    @Before
-    public void beforeMyModuleTests() {
-        entityManager = getHostContext().get(EntityManager.class);
-        runUntil(()-> getHostContext().get(MyModuleReadySystem.class).isMyModuleReady());
-    }
-    
     @Test
     public void testExample() {
-        WorldProvider worldProvider = getHostContext().get(WorldProvider.class);
-        BlockManager blockManager = getHostContext().get(BlockManager.class);
-
         // create some clients (the library connects them automatically)
-        Context clientContext1 = createClient();
-        Context clientContext2 = createClient();
+        Context clientContext1 = helper.createClient();
+        Context clientContext2 = helper.createClient();
+
+        // wait for both clients to be known to the server
+        helper.runUntil(()-> Lists.newArrayList(entityManager.getEntitiesWith(ClientComponent.class)).size() == 2);
+        Assertions.assertEquals(2, Lists.newArrayList(entityManager.getEntitiesWith(ClientComponent.class)).size());
 
-        // assert that both clients are known to the server
-        EntityManager hostEntityManager = getHostContext().get(EntityManager.class);
-        List<EntityRef> clientEntities = Lists.newArrayList(hostEntityManager.getEntitiesWith(ClientComponent.class));
-        Assert.assertEquals(2, clientEntities.size());
+        // run until a condition is true or until a timeout passes
```

Comment:
"run until condition is true" would mean that this would stop right away. I guess you either want to say

> run as long the condition holds or until the timeout passes

or 

> run until the condition is false or until a timeout passes